### PR TITLE
fix(cdk/tree): update active item on focus

### PR DESCRIPTION
**Pull request by zarend**
[fix(cdk/tree): update active item on focus](https://github.com/angular/components/pull/27908/commits/1c0149354a31f70626f3b0d4718fef6dd806f8b9) 

Update CdkTreeNode, TreeKeyManager and Tree "Load More" example
regarding progrmatic focus.

 - In CdkTreeNode, update the active item on the focus event.
 - expose TreeKeyManager.updateActiveItem
 - remove TreeKeyManager.onClick
 - In "Load More" example, when clicking "Load More", focus first node
   that is added.

### CODE REVIEW

**Comment by zarend** [comment]
Hi @BobobUnicorn ,

This is what I came up with for supporting programmatic focus.

The first approach I tried was finding a way to use CdkTreeNodeComponent.focus. That was difficult because I couldn’t find a way to get a reference to the CdkTreeNodeComponent of the “Load More” node. It seemed unnecessary for me to maintain a QueryList of tree node component when the tree already had one.

I ended up finding it easier to use DOM manipulation to find the dom node I wanted and call .focus on it.

I also remove the onClick method on TreeKeyManager so I wouldn’t have to add a duplicate method for onFocus. I exposed updateActiveItem to align with FocusKeyManager.

I also make a few changes to “Load More” example so we can test when there are two “Load More” nodes and I took care of misc small details in the demo.

WDYT?

-Zach

File: src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts [hunk 1]
```diff
@@ -132,57 +158,58 @@ export class TreeLoadmoreExample {
...
+      const previousSibling = nodeElement.previousElementSibling;
+
+      // Synchronously load data.
+      this._database.loadChildren(node.parent);
+
+      const focusDesination = previousSibling?.nextElementSibling || previousSibling;
+
+      if (focusDesination) {
+        // Restore focus.
+        (focusDesination as HTMLElement).focus();
```
Comment (BobobUnicorn) [inline src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts:211]: the one thing I'm concerned about here is whether or not the tabindex updates correctly.

Comment (zarend) [reply src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts:211]: I added a unit test to confirm that the tabindex updates.

Comment (zarend) [reply src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts:211]: nvm, that's something else. Let me double check that manually right now

Comment (zarend) [reply src/components-examples/material/tree/tree-loadmore/tree-loadmore-example.ts:211]: I tested the Load More example, it it correctly updates tabindex when clicking "Load More". It focuses the first new item that appears and sets it's tabindex to 0. All other tree nodes have tabindex="-1".

There's also a unit test to verify that CdkTreeNode correctly responds to focus event. Calling `.focus()` will trigger native focus event.

**Pull request merged**